### PR TITLE
impl(pubsub): add universe domain support to the publisher stub

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13293

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13310)

### CODE REVIEW

File: google/cloud/pubsub/internal/publisher_stub_factory.cc [hunk 1]
```diff
@@ -67,8 +70,22 @@ std::shared_ptr<PublisherStub> MakeRoundRobinPublisherStub(
 std::shared_ptr<PublisherStub> CreateDecoratedStubs(
     google::cloud::CompletionQueue cq, Options const& options,
     BasePublisherStubFactory const& base_factory) {
-  auto auth = google::cloud::internal::CreateAuthenticationStrategy(
-      std::move(cq), options);
+  auto endpoint = internal::DetermineServiceEndpoint(
+      internal::GetEnv("PUBSUB_ENDPOINT"),
+      internal::FetchOption<EndpointOption>(options), "pubsub.googleapis.com",
+      options);
```
Comment (alevenberg) [inline google/cloud/pubsub/internal/publisher_stub_factory.cc:76]: I don't really know what these values should be

Comment (alevenberg) [reply google/cloud/pubsub/internal/publisher_stub_factory.cc:76]: @scotthart ?

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13310)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @alevenberg and @scotthart)

___
*[`google/cloud/pubsub/internal/publisher_stub_factory.cc` line 76 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13310#gh-1427320538:-NliiF2J2n20PYzwia2h:b-7ymvsd) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/617134ce7e052e5458641b8855fd16fafac4d84e/google/cloud/pubsub/internal/publisher_stub_factory.cc#L76)):*
<details><summary><i>Previously, alevenberg (Anna Levenberg) wrote…</i></summary><blockquote>

@scotthart ?
</blockquote></details>

Looks like pubsub has multiple endpoints:
```
find google/cloud/pubsub -name '*.cc' -exec grep -H ENDPOINT {} \;
google/cloud/pubsub/admin/internal/subscription_admin_stub_factory.cc:      internal::GetEnv("GOOGLE_CLOUD_CPP_SUBSCRIBER_ENDPOINT"),
google/cloud/pubsub/admin/internal/topic_admin_stub_factory.cc:      internal::GetEnv("GOOGLE_CLOUD_CPP_PUBLISHER_ENDPOINT"),
google/cloud/pubsub/admin/internal/subscription_admin_option_defaults.cc:      std::move(options), "GOOGLE_CLOUD_CPP_SUBSCRIBER_ENDPOINT",
google/cloud/pubsub/admin/internal/topic_admin_option_defaults.cc:      std::move(options), "GOOGLE_CLOUD_CPP_PUBLISHER_ENDPOINT",
google/cloud/pubsub/internal/schema_stub_factory.cc:      internal::GetEnv("GOOGLE_CLOUD_CPP_SCHEMA_SERVICE_ENDPOINT"),
google/cloud/pubsub/internal/schema_option_defaults.cc:      std::move(options), "GOOGLE_CLOUD_CPP_SCHEMA_SERVICE_ENDPOINT",
```
We need to make sure that all of them can have the universe_domain value substituted for them.

File: google/cloud/pubsub/internal/publisher_stub_factory.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/publisher_stub_factory.cc:76]: Ack, fixed.

**Comment by alevenberg** [comment]
I think we just want the corresponding service endpoint to work. Updated it for the publisher service.

**Comment by dbolduc** [comment]
I think we should close this. `pubsub` has `UniverseDomainOption` support as of: #13334

**Comment by alevenberg** [comment]
I see... so we're making it a common option instead?

**Comment by dbolduc** [comment]
> I see... so we're making it a common option instead?

I am not sure what you mean. We are defaulting the `EndpointOption` to incorporate the `UniverseDomainOption` in a common place (`PopulateCommonOptions`), which `pubsub` already uses.

**Pull request closed without merging**